waku/v2/protocol/filter/client.go

```diff
@@ -306,7 +306,12 @@ func (wf *WakuFilterLightNode) Subscribe(ctx context.Context, contentFilter prot
 		cFilter.PubsubTopic = pubSubTopic
 		cFilter.ContentTopics = protocol.NewContentTopicSet(cTopics...)
 
-		err := wf.request(ctx, params, pb.FilterSubscribeRequest_SUBSCRIBE, cFilter)
+		err := wf.request(
+			ctx,
+			&FilterSubscribeParameters{selectedPeer: selectedPeer, requestID: params.requestID},
```

Comment:
Good catch, but rather than creating a new one manually like this, it would be better to copy the existing params and just set the selectedPeer in the new copied version and pass that. 
That way in future if new params are added, we don't need to manually pass them here, rather the copying would take care.

Reply:
Ok, it's upto you if you want to take it as separate PR or include in this.
But, hope you do get the maintainability problem i see with this.